### DISCUSSION TITLE
Docker image for building this project

### DISCUSSION THREAD

**Discussion by lifubang** [coder/code-server#5298] [Q&A] [answered]
I want to do some contribution to this great project according to the guide in `https://coder.com/docs/code-server/latest/CONTRIBUTING`, but I can't build it success, and I spent two day to try lots of ways, unfortunately, I still failed.
Could you please provide a `Docker image` or a `Dockerfile` for us to help building. Thanks.

**Comment by lifubang** [comment]
In env:
```
root@083944f78bbb:~/dev/code-server# uname -a
Linux 083944f78bbb 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
root@083944f78bbb:~/dev/code-server# node -v
v16.15.1
```
I got:
```
[10:36:20] Starting minify-vscode-reh-web ...
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
[10:36:32] Finished minify-vscode-reh-web after 12336 ms
[10:36:32] Starting node ...
[11:14:50] 'vscode-reh-web-linux-x64-min' errored after 46 min
[11:14:50] Error: aborted
    at connResetException (node:internal/errors:692:14)
    at TLSSocket.socketCloseListener (node:_http_client:414:19)
    at TLSSocket.emit (node:events:539:35)
    at TLSSocket.emit (node:domain:537:15)
    at node:net:709:12
    at TCP.done (node:_tls_wrap:582:7)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by lg-nineoneninethreenine** [comment, marked as answer]
There is a Dockerfile on the Docker Registry; if I am not mistaken, its image name has to be [`codercom/code-server:latest`](https://hub.docker.com/r/codercom/code-server). I hope this was what you were looking for.

**Reply by lifubang** [reply]
Thank you for your reply.

I think this is the Docker image of the release goods after `yarn package`, not the env for building this project from the source code.
I want to do some code change, and then do a new release. Just like this:
```
git clone https://github.com/coder/code-server.git
git submodule update --init
quilt push -a
yarn
yarn watch # Launch code-server localhost:8080. code-server will be live reloaded when changes are made; the browser needs to be refreshed manually.
```
You can try this flow, and you will fail in the step `yarn`.

**Reply by lifubang** [reply]
I think this may be some network errors.
All build work works fine, except I run `yarn watch`, I got 404 when I open `localhost:8080`:
```
ENOENT: no such file or directory, open '/home/acmcoder/dev/code-server/lib/vscode/out/vs/server/node/server.main.js'
```
And use `ls` to see:
```
acmcoder@acmcoder:~/dev/code-server$ ls lib/vscode/out/vs/server/node/
acmcoder@acmcoder:~/dev/code-server$
```
Exactly, this is no file in this folder.
Which step I was missing?
All steps is from here: `https://coder.com/docs/code-server/latest/CONTRIBUTING`

**Reply by lifubang** [reply]
All works fine now.
It needs a long time to complete the compilation even in a 8 * 16 workstation when running `yarn watch`.
